hledger-lib/Hledger/Data/Types.hs

```diff
@@ -299,7 +299,7 @@ data Posting = Posting {
       pdate2            :: Maybe Day,         -- ^ this posting's secondary date, if different from the transaction's
       pstatus           :: Status,
       paccount          :: AccountName,
-      pamount           :: MixedAmount,
+      pamount           :: [Amount],
```

Comment:
I approve of modelling what's needed more precisely with types. But is there not some loss of overall conceptual simplicity ? We tell users "a posting has a (possibly multicommodity) amount". This is still true on the outside, but developers now must deal with two different representations of multicommodity amounts (MixedAmount vs [Amount]). Could you explain/motivate more why it's better for Posting not to use a MixedAmount ? Maybe we could include it in the field's haddock. I do see that it seems to reduce code noise in many places. (Also, "pamount" singular now sounds a bit odd.)

Reply:
I originally tried using the same `MixedAmount` type for `pamount`, but it ended up breaking some of the tests. I seem to recall that it was in the close command.

That said, I started work on this part a long time ago, and perhaps the problem was not actually with the data representation, but another issue which was resolved elsewhere, or could be. I'll do another check to see whether that can be done.

Reply:
It turns out I was able to use MixedAmount for pamount.